server/src/server.ts

```diff
@@ -51,14 +51,15 @@ let currentActiveTextDocument: TextDocument = TextDocument.create(
 )
 
 connection.onInitialize(async (params: InitializeParams): Promise<InitializeResult> => {
-  logger.level = 'debug'
+  logger.level = params.initializationOptions?.loggingLevel ?? 'error'
   logger.info('[onInitialize] Initializing connection')
   bitBakeProjectScannerClient.setConnection(connection)
   disposables.push(...bitBakeProjectScannerClient.buildHandlers())
 
   workspaceFolder = params.workspaceFolders?.[0].uri.replace('file://', '')
 
-  const extensionPath = params.initializationOptions.extensionPath as string
+  const extensionPath = params.initializationOptions?.extensionPath ?? workspaceFolder

```

Comment:
workspaceFolder is very unlikely to match our extensionPath. Isn't there an alternative on VIM to get the extensionPath? Otherwise there's __dirname in Typescript which could be used?

Reply:
I am thinking about moving the docs to the server, then we'd never need to pass the extension path to the server.

Reply:
When initializationOptions don't have extensionPath, we can use workspace as a fallback. Whatever, we should'nt use **undefined** as a fallback.